applications/nrf_desktop/README.rst

```diff
@@ -1653,6 +1653,102 @@ For example, the following line will start the upload of the new image to the de
+
+Adding a new device
+-------------------
+
+The device specifies which protocol is used for the communication by adding the specific device information to the plugin metadata.
+For the ``nordic_hid`` plugin, this file is located at `nordic-hid.quirk`_.
+The following example shows the information passed to the plugin metadata:
+
+.. code-block:: console
+
+   [HIDRAW\VEN_1915&DEV_52DE]
+   Plugin = nordic_hid
+   GType = FuNordicHidCfgChannel
+   NordicHidBootloader = B0
+
+* ``[HIDRAW\VEN_1915&DEV_52DE]``: describes the method of communication (``HIDRAW`` selects communication using ``HID`` reports), vendor, and device identifiers.
+  This information tells the fwupd how to read the version number devices with this identifier and identifies the device.
+* ``Plugin = nordic_hid`` and ``GType = FuNordicHidCfgChannel``: sets the plugin that the device uses.
+* ``NordicHidBootloader``: selects the bootloader that the device is running.
+  ``NordicHidBootloader`` is an optional entry.
+  If the device does not have the information about the underlying bootloader, the ``NordicHidBootloader`` option is used to select a proper bootloader type.
+  If there is no information about the bootloader, both in metadata and from the device, the update procedure will fail.
+  The possible values are either ``B0`` or ``MCUBOOT``.
+
+.. note::
+   As the ``nordic_hid`` plugin communicates with the device using the Configuration channel, the device update is not allowed through the Serial recovery DFU.
+
+To add a new device, a pull request must be opened to the `fwupd`_ repository with a new entry to the :file:`nordic-hid.quirk` file.
+
+LVFS and update image preparation
+---------------------------------
+
+The `LVFS (Linux Vendor Firmware Serivce) <LVFS_>`_ hosts firmware images that can be downloaded by Linux machines and used by the fwupd tool for the firmware update of compatible devices.
+A vendor account is needed to upload a new firmware archive to the site.
+Information on how to apply for an account is found at `LVFS Getting an account`_ site.
+
+The nRF Desktop application DFU image is delivered as a zip package, containing a manifest and one or more binary files used for the update.
+To prepare the image file compatible with the LVFS, the CAB file needs to be prepared.
+The CAB package must contain the DFU package generated by the |NCS|, that is :file:`dfu_application.zip`, plus metadata file with information used by the LVFS.
+For more information, see `LVFS metadata`_ site.
+
+When the CAB archive has been built it can be uploaded to the LVFS where the CAB archive is verified and signed.
+For more information about creating CAB files, signing, and uploading the update package, see `LVFS Uploading Firmware`_ site.
+
+Firmware upgrade using fwupd
+----------------------------
+
+Once the update image was uploaded onto the LVFS the firmware update procedure can be tested on end machines.
```

Comment:
hm... I think here should be description how to test the CAB file locally prior uploading to LVFS.
I.e. something similar to:
```
fwupdtool install --plugins nordic_hid --verbose --allow-reinstall --allow-older nordic-nrf52840dk-mcuboot-0.0.0_2.cab
```

Reply:
maybe add a `sudo` in front of both of those for clarity that they need to be run as root?